x-pack/plugins/enterprise_search/server/routes/enterprise_search/crawler/crawler.ts

```diff
@@ -23,71 +17,21 @@ import { registerCrawlerSitemapRoutes } from './crawler_sitemaps';
 export function registerCrawlerRoutes(routeDependencies: RouteDependencies) {
   const { router, enterpriseSearchRequestHandler, log } = routeDependencies;
 
-  router.post(
+  router.put(
```

Comment:
suggestion: we should keep this as a post with an indexName, if you ask me. We're not updating an existing crawler, we're creating a crawler.

Reply:
@sphilipse we aren't creating the crawler index in ent-search anymore. The creation happens at the connectors endpoint. The crawler endpoint now adds crawler-specific settings/mappings and creates related indices like configuration. You can see the changes here: https://github.com/elastic/ent-search/pull/6786

Reply:
The connectors endpoint creates a connector with a 'crawler' service_type (the way you've implemented it right now). But that's just a document in an index, it's not a crawler until you've done this setup--and this setup is not updating anything, it's creating new settings, mappings and related indices entries.  

Think of it this way: if you create a database entry somewhere, you'd do a POST request. If you then add a bunch of separate *new* entries in different database tables, you're still doing POST requests--you're not updating existing objects, you're creating new ones. 

But this logic being split doesn't make much sense, and why I suggested to merge it: you can't have a crawler connector document without a crawler infrastructure elsewhere, or vice versa. So this should all live as a single endpoint that does all of that creation.

Reply:
^ discussed in a call, will move the API call to create a connector to the kibana server side